### DISCUSSION TITLE
Is there a plan to support jwt sign with kid in header

### DISCUSSION THREAD

**Discussion by byn9826** [honojs/hono#3397] [Q&A]
Currently the header of jwt sign is fixed, and consist of { alg, typ: 'JWT' }  https://github.com/honojs/hono/blob/f9a23a9992979fed79b0703ab8b3a3ce49f7175f/src/utils/jwt/jwt.ts#L53

To support adding kid in the header during signing, I’ve had to copy all the helper functions from Hono’s JWT sign functionality, allowing me to set kid in the header manually. This approach is definitely not super clean and duplicating codes:
https://github.com/ValueMelody/melody-auth/blob/main/server/src/services/jwt.ts#L52

Since including kid in the JWT header is a common practice when signing tokens with RS256, I’m wondering if there are any plans to officially support it directly in Hono’s JWT sign function.

**Comment by ssadel** [comment]
bump, need this for sign in with apple server side validation

**Comment by DevTurpGL** [comment]
I'd like this, too.  Surprised it wasn't added in the beginning.

**Comment by AnuroopSuresh** [comment]
any update on this?
